cranelift/assembler-x64/src/vex.rs

```diff
@@ -0,0 +1,339 @@
+//! Encoding logic for VEX instructions.
+//use super::evex::{Register, RegisterOrAmode};
+
+use super::rex;
+use super::XmmMem;
+use crate::api::{AsReg, CodeSink, KnownOffsetTable, Registers};
+use crate::mem::emit_modrm_sib_disp;
+use crate::Amode;
+use cranelift_assembler_x64_meta::dsl::{OpcodeMap, Vex};
```

Comment:
Would it be possible to duplicate necessary types here? The intent is that the meta crate is only used at build-time and isn't a runtime dependency but this is otherwise adding the meta crate as a runtime dependency as well

Reply:
Hi @alexcrichton, I think so. I did it this way as it seemed to be required for a previous iteration of this patch set, plus it assures consistency and avoids duplication, but I understand we don't want a dependency on the meta crate and that duplication is ok. If I do believe there is a dependency we want to keep, I'll follow @cfallin advise and see if a shared crate makes sense here. Also note, there are a few unnecessary elements in the vex struct in the meta crate that I am removing along with commented out code intended to help develop the follow-up patch 2 of 2 to be submitted later when we enable this.

Reply:
@alexcrichton .. I believe this is addressed now.